### PR TITLE
Source and Sink sacura test

### PR DESCRIPTION
**Pull request by pierDipi**
Part of #312

Add a sacura test using KafkaSink + KafkaSource with a "slow" consumer.

This test has been useful to catch bugs that are now fixed

### CODE REVIEW

**Comment by pierDipi** [comment]
No duplicates and expected time: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/1417/reconciler-tests_eventing-kafka-broker_main/1551979818151579648


/test reconciler-tests_eventing-kafka-broker_main
/test integration-tests_eventing-kafka-broker_main

**Comment by pierDipi** [comment]
Another one with no duplicates https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/1417/reconciler-tests_eventing-kafka-broker_main/1552189381056401408

**Comment by pierDipi** [comment]
No duplicates and expected time (you need to scroll the test/e2e: TestSacuraSinkSourceJob logs) https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/1417/integration-tests_eventing-kafka-broker_main/1552214386783817728

**Comment by aslom** [comment]
@pierDipi @aavarghese  are those test include re-balancing? any scaling up or down?

**Comment by pierDipi** [comment]
> @pierDipi @aavarghese are those test include re-balancing? any scaling up or down?

There is no scale up or down for now.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/OrderedAsyncExecutor.java [hunk 1]
```diff
@@ -37,8 +38,8 @@ public class OrderedAsyncExecutor {
 
   private final Queue<Task> queue;
 
-  private boolean isStopped;
-  private boolean inFlight;
+  private final AtomicBoolean isStopped;
```
Comment (matzew) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/OrderedAsyncExecutor.java:41]: was this change _needed_, or was it just handy, while at it?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**